### PR TITLE
Work on a few issues...

### PR DESCRIPTION
**Pull request by jogibear9988**
I've started to work on a few issues, but not yet found time to test against real s7

### CODE REVIEW

**Comment by jogibear9988** [comment]
maybe the byte order in s5time could be wrong...

File: main.js [hunk 1]
```diff
@@ -534,6 +530,27 @@ function updateConnection(_connected) {
...
+    {
+        case 'WORD':
+        case 'INT':
+        case 'S5TIME':
+            return 2;
+        case 'DWORD':
+        case 'DINT':
+        case 'REAL':
+            return 4;
+        case 'DWORD':
```
Comment (Apollon77) [inline main.js:544]: DWORD is there twice

File: main.js [hunk 2]
```diff
@@ -534,6 +530,27 @@ function updateConnection(_connected) {
...
+            return 8;
+        case 'DWORD':
```
Comment (Apollon77) [inline main.js:546]: Copy paste error? STRING, ARRA and S7STRING missing

File: main.js (hunk 1 above)
Comment (Apollon77) [reply main.js:544]: should be S7TIME?

File: main.js [hunk 3]
```diff
@@ -1084,6 +1053,9 @@ const main = {
         if (!s7client) {
             return;
         }
+        if (main.acp.connectionType) {
```
Comment (Apollon77) [inline main.js:1056]: Where this is coming from? there is no config option for this or?!

**Review by Apollon77 — CHANGES_REQUESTED** [review]

**Comment by Apollon77** [comment]
@GermanBluefox This was not a good idea unless you want to fix now all tghe issues introduced by this PR :-(( See my comments ... there are several things completely unclear with the PR

**Comment by Apollon77** [comment]
Ok seem some fixes were done, but https://github.com/ioBroker/ioBroker.s7/blob/master/main.js#L1053 is still in ...

**Comment by jogibear9988** [comment]
Never saw the comments...
The connectionType has no UI but you can modify via objects.
Most people 99% normaly don't need it.

And no it is S5Time not S7Time

**Comment by Apollon77** [comment]
>  The connectionType has no UI but you can modify via objects.

Undocumented settings ? Without a Code comment for the purpose and why it is undocumented setting and for what to use it OR Readme  info as "expert stuff" OR anything else that helps understand it, it will be seen by any developer (including you maybe in a year) as wrong and buggy code. my opinion about it.

Then please have a look at the fixes @GermanBluefox did today and check if he fixed anything correctly and create a PR if not because it is already released now :-(

**Comment by jogibear9988** [comment]
I looked at the fixes from BlueFox, they look good atm.

Problem is, there is a S7Time defined in the Code, but on PLC Side the type meant is "DATE_AND_TIME"
But if we rename it, it would be a breaking change

There are also more types missing like "Time", "Time_Of_Day", ....
Or newer ones from 1500 PLCs like LReal, LInt,...

**Comment by Apollon77** [comment]
Can you please provide a PR with some code comments for that undocumented option please that we at least have infos in code? Thank you

**Comment by jogibear9988** [comment]
What infos do you want? It's only one line of code, wich uses the value from the settings to call the api from snap7.
Should we list the constants?

0x01 = PG connection
0x02 = OP connection
0x03 = Other connection
0xFD = S7 Basic connection

**Comment by Apollon77** [comment]
I would like to see something helpful that describes why this code part https://github.com/ioBroker/ioBroker.s7/blob/master/main.js#L1053 exists and in which cases the user can use this for what. Somethig that describes other developers what this special code does and why it is there.

```
// Use this unofficial parameter to set the Connection type if you need a different one then the default (**Add what default is**)
```

Yes also the allowed values, or maybe a link to the snap7 specs for it, can also help

But effectively ... I still do not understand why we need an unofficial parameter here and not simply add it correctly as an adapter parameter ...

**Pull request merged**